src/screens/FilterModal/FilterModal.tsx

```diff
@@ -24,7 +27,7 @@ export const FilterModal: React.FC<{
   const dispatch = useAppDispatch();
   const activeFilter = useAppSelector((state) => state.filters.activeFilter);
   const logsByTag = useAppSelector((state) => state.filters.logsByTag);
-  const filterByPic = useAppSelector((state) => state.filters.filterByPic);
+  const filteringByPic = useAppSelector((state) => state.filters.logsWithPics);
   const logsByDate = useAppSelector((state) => state.filters.logsBydate);
   const [selectedTags, setSelectedTags] = useState<string[]>([]);
   const logsByLocation = useAppSelector(
```

Comment:
You could wrap these up in a selector within the slice
then it'd be something like 
```ts
// the api would end up being something like this
const  {filterByPick, logsByTag, .....} = useAppSelector(selectFilters)
```

another pattern I've seen a bit is this, declaring a selector within the component file
```ts
const  {filterByPic, logsByTag, .....} = useSelector(mapStateToProps)

// you can declare this outside the component within the same file
function mapStateToProps(state: RootState) {
 return {
   filterByPic: state.filters.filterByPic,
   logsByTag:  state.filters.filterByPic(state),
  // ...other state maps
 }
}
```
 though its just a selector and probably better to move that to the slice, I generally only see it like this when there's multiple state selects that aren't related.

Reply:
Similar to useState, selectors are a great place to derive other values from your state so your state can be kept to a minimum and extra state values don't need to be added and managed. 

example
```ts
// derived state
const [todos, setTodos]  = useState([
  {body: 'get milk', completed: false},
  {body: 'lick dicks', completed: true},
]]

const completedTodos = todos.filter(todo => todo.completed)

// duplicating state
const [todos, setTodos] = useState([
  {body: 'get milk', completed: false},
  {body: 'lick dicks', completed: true},
]]

const [completedTodos, setCompletedTodos]  = useState([])
useEffect(() => {
// ... some set logic. or could be in some handler
}, [todos])


``` 

see https://redux.js.org/usage/deriving-data-selectors